Port AccessibilityPreferences & InsertTextOptions to the new serialization format

gavin-apple opened the pull request:
#### eb332de881aca457c9fdf2542bea8d20d38c5f3b
<pre>
Port AccessibilityPreferences &amp; InsertTextOptions to the new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=262645">https://bugs.webkit.org/show_bug.cgi?id=262645</a>
rdar://116491012

Reviewed by Alex Christensen.

This change ports AccessibilityPreferences, InsertTextOptions &amp; AXValueState
to the new IPC serialization format.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/Shared/AccessibilityPreferences.cpp: Removed.
* Source/WebKit/Shared/AccessibilityPreferences.h:
* Source/WebKit/Shared/AccessibilityPreferences.serialization.in: Added.
* Source/WebKit/Shared/Cocoa/InsertTextOptions.cpp: Removed.
* Source/WebKit/Shared/Cocoa/InsertTextOptions.h:
* Source/WebKit/Shared/Cocoa/InsertTextOptions.serialization.in: Added.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/269004@main">https://commits.webkit.org/269004@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d875ca429281e30f890db12b007e8087f74486d4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21294 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21624 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22347 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23158 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/19759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24904 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21853 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20973 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21517 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18458 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24013 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18356 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25643 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19429 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19507 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23482 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/20008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19310 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23571 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2636 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19897 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
```diff
@@ -30,12 +30,12 @@
 
 #if !USE(APPLE_INTERNAL_SDK)
-typedef enum {
+enum class AXValueState : int8_t {
```
This seems like it would no longer build with the internal SDK.  Does it?

achristensen07 replied:
Revert this change and it's good.

achristensen07 commented on Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
```diff
@@ -211,3 +211,12 @@ enum class WTFLogLevel : uint8_t {
 };
+
+#if HAVE(PER_APP_ACCESSIBILITY_PREFERENCES)
+enum class AXValueState : int8_t {
```
This needs to be an int now.

gavin-apple committed:
Port AccessibilityPreferences & InsertTextOptions to the new serialization format
https://bugs.webkit.org/show_bug.cgi?id=262645
rdar://116491012

Reviewed by Alex Christensen.

This change ports AccessibilityPreferences, InsertTextOptions & AXValueState
to the new IPC serialization format.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/spi/Cocoa/AccessibilitySupportSPI.h:
* Source/WebKit/PlatformMac.cmake:
* Source/WebKit/Shared/AccessibilityPreferences.cpp: Removed.
* Source/WebKit/Shared/AccessibilityPreferences.h:
* Source/WebKit/Shared/AccessibilityPreferences.serialization.in: Added.
* Source/WebKit/Shared/Cocoa/InsertTextOptions.cpp: Removed.
* Source/WebKit/Shared/Cocoa/InsertTextOptions.h:
* Source/WebKit/Shared/Cocoa/InsertTextOptions.serialization.in: Added.
* Source/WebKit/Shared/WTFArgumentCoders.serialization.in:
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/269004@main
